src/components/ui/Transactions/Transactions.tsx

```diff
@@ -0,0 +1,115 @@
+import { Currency } from '../../../api/types/Enums';
+import { PaymentMethod } from '../../../types/PaymentMethodsEnum';
+import CardIcon from '../../../assets/icons/card-icon.svg';
+import BankIcon from '../../../assets/icons/bank-icon.svg';
+import WalletIcon from '../../../assets/icons/wallet-icon.svg';
+import { useState } from 'react';
+import { AnimatePresence, motion } from 'framer-motion';
+
+export interface Transaction {
+  occurredAt: Date;
+  paymentMethod: PaymentMethod;
+  amount: number;
+  currency: Currency.EUR | Currency.GBP;
+  processor: string;
+  last4DigitsOfCardNumber?: string;
+}
+
+const PaymentMethodIcon = ({ paymentMethod }: { paymentMethod: PaymentMethod }) => {
+  switch (paymentMethod) {
+    case PaymentMethod.Card:
+      return <img src={CardIcon} alt="card" />;
+    case PaymentMethod.Pisp:
+      return <img src={BankIcon} alt="bank" />;
+    case PaymentMethod.ApplePay:
+    case PaymentMethod.GooglePay:
+      return <img src={WalletIcon} alt="wallet" />;
+    default:
+      return null;
+  }
+};
+
+const formatDate = (date: Date): string => {
+  const options: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' };
+  const formattedDate = new Date(date).toLocaleString('en-US', options);
+
+  const day = new Date(date).getDate();
+  const daySuffix = getDaySuffix(day);
+
+  return `${formattedDate}${daySuffix}, ${new Date(date).getFullYear()}`;
+};
+
+const getDaySuffix = (day: number) => {
+  if (day >= 11 && day <= 13) {
+    return 'th';
+  }
+
+  switch (day % 10) {
+    case 1:
+      return 'st';
+    case 2:
+      return 'nd';
+    case 3:
+      return 'rd';
+    default:
+      return 'th';
+  }
+};
+const Transactions = ({ transactions }: { transactions: Transaction[] }) => {
+  const [hoveredRow, setHoveredRow] = useState<number | null>(null);
+
+  return (
+    <div className="px-4 pb-4 ">
+      <table>
+        <tbody>
+          {transactions.map((transaction, index) => (
+            <tr
```

Comment:
I reckon it'd be a smart move to set up the Transaction component separately. We're gonna need to include a `onRefundClick` property or something along those lines to manage that event.

If you could add that event, that'd be awesome. And if you end up making a standalone Transaction component, could you also add a story for it (as in Storybook story)?

Reply:
I have added the handleRefund method in the component which takes the paymentAttemptID